src/go-slang/goroutine.ts

```diff
@@ -293,14 +298,43 @@ const Interpreter: {
   BinaryExpression: ({ left, right, operator: op }: BinaryExpression, { C, H, A }) =>
     C.pushR(...H.allocM([left, right, { type: CommandType.BinaryOp, opNodeId: A.track(op).uid }])),
 
-  CallExpression: ({ callee, args }: CallExpression, { C, H, A }) =>
-    C.pushR(
-      ...H.allocM([
-        callee,
-        ...args,
-        { type: CommandType.CallOp, calleeNodeId: A.track(callee).uid, arity: args.length }
-      ])
-    ),
+  CallExpression: ({ callee, args }: CallExpression, { C, E, S, H, A }) => {
+    if (callee.type !== NodeType.QualifiedIdentifier) {
+      C.pushR(
+        ...H.allocM([
+          callee,
+          ...args,
+          { type: CommandType.CallOp, calleeNodeId: A.track(callee).uid, arity: args.length }
+        ])
+      )
+      return
+    }
+
+    const className = H.resolve(E.lookup(callee.pkg.name))
+    if (className === null) {
+      return Result.fail(new UndefinedError(callee.pkg.name, callee.loc!))
+    }
+
+    if (className instanceof WaitGroup) {
+      const methodActions = {
+        Add: () => ({ type: CommandType.WaitGroupAddOp, count: (args[0] as Literal).value }),
+        Done: () => ({ type: CommandType.WaitGroupDoneOp }),
+        Wait: () => ({ type: CommandType.WaitGroupWaitOp })
+      }
+
+      const action = methodActions[callee.method.name]()
```

Comment:
We will need to check if `callee.method.name` exists in the `methodActions.keys()` first. In the case where name does not exist, we will be calling a non-function